### DISCUSSION TITLE
Doc versioning + ReactLiveScope

### DISCUSSION THREAD

**Discussion by M1chaelChen** [facebook/docusaurus#7272] [Q&A]
Hi, we have a regular doc site setup with ReactLiveScope that imports the design system packages we built. Now we are planning to add [versioning](https://docusaurus.io/docs/versioning) but then realize that the packages imported in ReactLiveScope will all point to the same version. Is there a way to configure different live scope package versions for different doc versions? For example, for doc version 1.1.0, I want live scope to use React 16, and for doc version 1.2.0, I want it to use React 18.
Would appreciate any help here.

**Comment by slorber** [comment]
Hi

Unfortunately I don't think it is easy to do that, and you'd be better using a different site per lib version (you don't have to follow the 1 SPA for multiple versions that Docusaurus uses by default, see how older Docusaurus versions are archived as separate sites for example)


Note: React < 17 is not really designed to be run alongside other React versions in the same app

See also https://reactjs.org/blog/2020/08/10/react-v17-rc.html#changes-to-event-delegation
